docs/metrics.md

```diff
@@ -127,3 +127,17 @@ Count*. The count metrics are illustrated by the following examples:
   the server. *Request Count* = 2, *Inference Count* = 9, *Execution
   Count* = 1.
 
+## Custom Metrics
+
+Triton exposes a C API to allow users and backends to register and collect
+custom metrics with the existing Triton metrics endpoint. It is up to the user
+to manage the lifetime of their custom metrics via the C API, Triton will not
```

Comment:
something like "The user takes the ownership of the custom metrics created through the APIs and must manage their lifetime following the API documentation"?